[GEOS-10469] tidy up logging instructions (#5861)
geoserver/geoserver: doc/en/user/source/production/troubleshooting.rst

Looking into such file is one of the first things to do when troubleshooting a problem, in particular it's interesting to see the log contents in correspondence of a misbehaving request.
The amount of information logged can vary based on the logging profile chosen in the *Server Settings* configuration page.

.. _troubleshooting_requests:

Logging service requests
------------------------

GeoServer provides a request logging capability that is normally inactive. When enabled in the :ref:`global settings <config_globalsettings_log_request>` GeoServer can log both the requested URL and POST requests contents.

> These settings are disabled by default due to their overhead. If you need a history of the incoming requests enable request logging by navigating to :menuselection:`Settings > Global` page, and scroll down to **Logging Settings** to :ref:`Enable Request Logging <config_globalsettings_log_request>``. You can enable this feature using *Enable Request Logging* (and optionally *Log Request Bodies* and *Log Request Headers*) before clicking :guilable:`Apply` as shown below.

Reviewer: ```suggestion
These settings are disabled by default due to their overhead. If you need a history of the incoming requests enable request logging by navigating to :menuselection:`Settings > Global` page, and scroll down to **Logging Settings** to :ref:`Enable Request Logging <config_globalsettings_log_request>``. You can enable this feature using *Enable Request Logging* (and optionally *Log Request Bodies* and *Log Request Headers*) before clicking :guilabel:`Apply` as shown below.
```

> Normally it is disabled due to its overhead. If you need to have an history of the incoming requests you navigate to :menuselection:`Settings > Global` page under *Internal Settings* *Logging Settings* as shown below. Enable logging service requests by selecting :guilabel:`Enable Request Logging` (and optionally :guilabel:`Log Request Bodies` and :guilabel:`Log Request Headers``) before clicking :guilabel:`Apply`.

Reviewer 2: most of this sentence duplicates the previous sentence

jps
````

> `jps <http://java.sun.com/javase/6/docs/technotes/tools/share/jps.html>`__ is a tool listing all the Java processing running. It can be used to retried the ``pid`` (process id) of the virtual machine that is running GeoServer. For example::

Reviewer 2: ```suggestion
`jps <http://java.sun.com/javase/6/docs/technotes/tools/share/jps.html>`__ is a tool listing all the Java processing running. It can be used to retrieve the ``pid`` (process id) of the virtual machine that is running GeoServer. For example::
```

   11521  -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -Djava.library.path=/usr/lib/jni -Dosgi.requiredJavaVersion=1.5 -XX:MaxPermSize=256m -Xms64m -Xmx1024m -XX:CMSClassUnloadingEnabled -XX:CMSPermGenSweepingEnabled -XX:+UseParNewGC
   16287 sun.tools.jps.Jps -mlv -Dapplication.home=/usr/lib/jvm/java-6-sun-1.6.0.16 -Xms8m
	
The output shows the ``pid``, the main class name if available, and the parameters passed to the JVM at startup. In this example ``16235`` is Tomcat hosting GeoServer, ``11521`` is an Eclipse instance, and ``16287`` is ``jps`` itself. In the common case you'll have only few JVM and the one running GeoServer can be identified by the parameters passed to it.

jstack
````````

`jstack <http://java.sun.com/javase/6/docs/technotes/tools/share/jstack.html>`__ is a tool extracting a the current stack trace for each thread running in the virtual machine. It can be used to identify scalability issues and to gather what the program is actually doing. 

> It usually takes people knowing about the inner workings of GeoServer can properly interpret the jstack output.

Reviewer: ```suggestion
It usually requires detailed understanding of the inner workings of GeoServer to properly interpret the jstack output.
```

    - org.apache.jk.common.ChannelSocket.accept(org.apache.jk.core.MsgContext) @bci=46, line=306 (Interpreted frame)
    - org.apache.jk.common.ChannelSocket.acceptConnections() @bci=72, line=660 (Interpreted frame)
    - org.apache.jk.common.ChannelSocket$SocketAcceptor.runIt(java.lang.Object[]) @bci=4, line=870 (Interpreted frame)
    - org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run() @bci=167, line=690 (Interpreted frame)
    - java.lang.Thread.run() @bci=11, line=619 (Interpreted frame)
   
   Locked ownable synchronizers:
       - None
   
   Thread 16267: (state = BLOCKED)
    - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
    - java.lang.Object.wait() @bci=2, line=485 (Interpreted frame)
    - org.apache.tomcat.util.threads.ThreadPool$ControlRunnable.run() @bci=26, line=662 (Interpreted frame)
    - java.lang.Thread.run() @bci=11, line=619 (Interpreted frame)
   
   Locked ownable synchronizers:
       - None
       
   ...
  
  
jmap
````

`jmap <http://java.sun.com/javase/6/docs/technotes/tools/share/jmap.html>`__ is a tool to gather information about the a Java virtual machine. 
It can be used in a few interesting ways.

> By running it without arguments (past the pid of the JVM) it will print out a **dump of the native libraries used by the JVM**. This can come in handy when one wants to double check GeoServer is actually using a certain version of a native library (e.g., GDAL)::

Reviewer: ```suggestion
By running it without arguments (other than the process id of the JVM) it will print out a **dump of the native libraries used by the JVM**. This can come in handy when one wants to double check GeoServer is actually using a certain version of a native library (e.g., GDAL)::
```
